prinz/src/main/scala/pl/touk/nussknacker/prinz/util/repository/client/RepositoryClientFactory.scala

```diff
@@ -0,0 +1,30 @@
+package pl.touk.nussknacker.prinz.util.repository.client
+
+import pl.touk.nussknacker.prinz.model.repository.ModelRepositoryException
+import pl.touk.nussknacker.prinz.util.repository.payload.ModelPayload
+
+import java.io.InputStream
+import java.net.URI
+
+class RepositoryClientFactory(implicit config: RepositoryClientConfig) {
+
+  def listModelFiles(path: URI): Either[ModelRepositoryException, Iterable[ModelPayload]] =
+    getClient(path).listModelFiles(path)
+
+  def openModelFile(path: URI): InputStream = getClient(path).openModelFile(path)
```

Comment:
these two methods shouldn't be in factory of client but in client.
Additionally the getClient method should be called usually once per repository creation (not on every repository method invoke)